android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp

```diff
@@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
 {
   return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
 }
+
+JNIEXPORT jstring JNICALL
+Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
+     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
+{
+  Bookmark const * bmk = getBookmark(catId, bmkId);
+  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
```

Comment:
Я имел в виду, что GetTitle для букмарок всё ещё не работает, например. Про GetAddress - слышу впервые, сейчас пытаюсь пофиксить важные баги, не уследил за изменением place_page::Info.

> On Mar 25, 2016, at 5:27 PM, Alex Zolotarev <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/bookmarks/data/Bookmark.cpp <https://github.com/mapsme/omim/pull/2526#discussion_r57448219>:
> 
> > @@ -81,4 +81,13 @@ Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeEncode2Ge0Url(
> >  {
> >    return jni::ToJavaString(env, frm()->CodeGe0url(getBookmark(cat, bmk), addName));
> >  }
> > +
> > +JNIEXPORT jstring JNICALL
> > +Java_com_mapswithme_maps_bookmarks_data_Bookmark_nativeGetAddress(
> > +     JNIEnv * env, jobject thiz, jint catId, jlong bmkId)
> > +{
> > +  Bookmark const * bmk = getBookmark(catId, bmkId);
> > +  search::AddressInfo const address = g_framework->NativeFramework()->GetAddressInfoAtPoint(MercatorBounds::FromLatLon(bmk->GetLatLon()));
> Это значит, что когда отображается pp, у него вызывается GetAddress и выводится на экран. Что значит не дорефакторен pp для букмарок? Он хранит айдишник, работа с айдишником идет через другую апи. Ты предлагаешь ее продублировать и в pp? Но зачем?
> … <x-msg://22/#>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mapsme/omim/pull/2526/files/c026c407dde3767df58339f1a8a905cbb630eeba#r57448219>